Documentation/docs.md

```diff
@@ -47,6 +47,7 @@ To learn more about the concepts and internals behind etcd, read the following p
  - [Why etcd][why] (TODO)
  - [Understand data model][data_model]
  - [Understand APIs][understand_apis]
+ - [Understand the design of the Auth subsystem][understand_auth]
```

Comment:
something like this?

* Why etcd?
* Data model
* APIs
* Internals
  * Raft
  * Write-ahead log
  * Key-value backend
  * Auth subsystem